### PR TITLE
Integrate a database model

### PR DESCRIPTION
**Pull request by chacaa**
Integrates a data base to the app (containing all the necessary models)

### Trello Card:
* [Create database and models](https://trello.com/c/oMIlPdCI)

/cc @xmartlabs/android

### CODE REVIEW

File: app/src/androidTest/java/com/xmartlabs/doapp/TestApplication.java [hunk 1]
```diff
@@ -19,4 +19,4 @@ protected ApplicationComponent createComponent() {
   protected ObjectGraph createBullet(ApplicationComponent component) {
     return new BulletTestComponent((TestComponent) component);
   }
-}
+}
```
Comment (mirland) [inline app/src/androidTest/java/com/xmartlabs/doapp/TestApplication.java]: add new line

File: app/src/main/java/com/xmartlabs/doapp/Gender.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.xmartlabs.doapp;
+
+/**
+ * Created by scasas on 3/7/17.
+ */
+
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/Gender.java]: remove line

File: app/src/main/java/com/xmartlabs/doapp/Gender.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+public enum Gender {
+  MALE,
+  FEMALE,
+  OTHER;
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/Gender.java]: add a comma, and put the semicolon in a new line

File: app/src/main/java/com/xmartlabs/doapp/model/Group.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * Created by scasas on 3/7/17.
+ */
+@AllArgsConstructor
+@Data
+@Builder
+@NoArgsConstructor
+@Parcel
+@Table(database = AppDataBase.class, cachingEnabled = true, cacheSize = 100)
+public class Group extends BaseModel {
+  @PrimaryKey
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/model/Group.java]: shouldn't be auto-incremented?

File: app/src/androidTest/java/com/xmartlabs/doapp/TestApplication.java (hunk 1 above)
Comment (diegomedina248) [inline app/src/androidTest/java/com/xmartlabs/doapp/TestApplication.java]: end line

File: app/src/main/java/com/xmartlabs/doapp/model/Task.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  long id;
+  @Column
+  String name;
+  @Column
+  Date date;
+  @Column
+  Double duration;
+  @Column
+  boolean isFinished;
+  @ForeignKey
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/model/Task.java:38]: did you evaluate using `@ForeignKeyContainer`?

File: app/src/androidTest/java/com/xmartlabs/doapp/TestApplication.java (hunk 1 above)
Comment (bryant1410) [inline app/src/androidTest/java/com/xmartlabs/doapp/TestApplication.java]: Why?

File: app/src/main/java/com/xmartlabs/doapp/Gender.java (hunk 3 above)
Comment (bryant1410) [inline app/src/main/java/com/xmartlabs/doapp/Gender.java]: Remove the semicolon

Comment (diegomedina248) [inline app/src/main/java/com/xmartlabs/doapp/Gender.java]: Add a comma instead.
The `;` is not needed unless you're adding code below

File: app/src/main/java/com/xmartlabs/doapp/model/Group.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * Created by scasas on 3/7/17.
+ */
+@AllArgsConstructor
+@Data
+@Builder
```
Comment (diegomedina248) [inline app/src/main/java/com/xmartlabs/doapp/model/Group.java]: alpha

File: app/src/main/java/com/xmartlabs/doapp/model/Task.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * Created by scasas on 3/7/17.
+ */
+@AllArgsConstructor
+@Data
+@Builder
```
Comment (diegomedina248) [inline app/src/main/java/com/xmartlabs/doapp/model/Task.java]: same here

File: app/src/main/java/com/xmartlabs/doapp/model/Task.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+@NoArgsConstructor
+@Parcel
+@Table(database = AppDataBase.class, cachingEnabled = true, cacheSize = 100)
+public class Task extends BaseModel {
+  @PrimaryKey(autoincrement = true)
+  long id;
+  @Column
+  String name;
+  @Column
+  Date date;
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/model/Task.java]: use `LocalDateTime` (the new date!)

File: app/src/main/java/com/xmartlabs/doapp/model/User.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * Created by scasas on 3/7/17.
+ */
+@AllArgsConstructor
+@Data
+@Builder
```
Comment (diegomedina248) [inline app/src/main/java/com/xmartlabs/doapp/model/User.java]: here too

**Review by bryant1410 — CHANGES_REQUESTED** [review]
I think you should add a Database class specifying the version.

Also, you can setup a CI server if you want

File: app/src/main/java/com/xmartlabs/doapp/model/User.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  @PrimaryKey(autoincrement = true)
+  long id;
+  @Column
+  String name;
+  @Column
+  Gender gender;
+  @Column
+  String email;
+  @Column
+  Date date;
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/model/User.java]: use LDT too

**Review by diegomedina248 — APPROVED** [review]
Left some comments

**Review by mirland — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/xmartlabs/doapp/TestApplication.java (hunk 1 above)
Comment (michaelEllisUy) [inline app/src/androidTest/java/com/xmartlabs/doapp/TestApplication.java]: end of line

File: app/src/main/java/com/xmartlabs/doapp/Gender.java (hunk 3 above)
Comment (michaelEllisUy) [inline app/src/main/java/com/xmartlabs/doapp/Gender.java]: remove semicolon, add a coma instead

**Review by michaelEllisUy — APPROVED** [review]

File: app/src/main/java/com/xmartlabs/doapp/Gender.java (hunk 3 above)
Comment (mirland) [reply app/src/main/java/com/xmartlabs/doapp/Gender.java]: yes, mike is right! the semicolon is not necessary

File: app/src/main/java/com/xmartlabs/doapp/model/Task.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+@NoArgsConstructor
+@Parcel
+@Table(database = AppDataBase.class, cachingEnabled = true, cacheSize = 100)
+public class Task extends BaseModel {
+  @PrimaryKey(autoincrement = true)
+  long id;
+  @Column
+  String name;
+  @Column
+  LocalDateTime date;
```
Comment (mirland) [inline app/src/main/java/com/xmartlabs/doapp/model/Task.java:33]: shouldn't you need a converter?

**Review by mirland — APPROVED** [review]

**Pull request merged**